org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/jobs/SonarLintMarkerUpdater.java

```diff
@@ -366,41 +368,41 @@ private static void createQuickFixMarkersForLocalIssues(IDocument document, ISon
       return;
     }
     var qfs = new ArrayList<MarkerQuickFix>();
-    for (var engineQuickFix : trackedIssue.getIssueFromAnalysis().quickFixes()) {
+    for (var engineQuickFix : trackedIssue.getIssueFromAnalysis().getQuickFixes()) {
       createQuickFix(document, issuable, qfs, engineQuickFix);
     }
     marker.setAttribute(MarkerUtils.SONAR_MARKER_QUICK_FIXES_ATTR, new MarkerQuickFixes(qfs));
   }
 
-  private static void createQuickFix(IDocument document, ISonarLintIssuable issuable, List<MarkerQuickFix> qfs, QuickFix engineQuickFix) {
-    var qf = new MarkerQuickFix(engineQuickFix.message());
-    for (var edits : engineQuickFix.inputFileEdits()) {
-      var inputFile = (EclipseInputFile) edits.target();
-      if (!issuable.equals(inputFile.getClientObject())) {
-        SonarLintLogger.get().debug("Quick fix on multiple files is not supported yet: " + engineQuickFix.message());
-        return;
-      }
-      if (inputFile.hasDocumentOlderThan(document)) {
-        SonarLintLogger.get().debug("Document has changed since quick fix was contributed: " + engineQuickFix.message());
+  private static void createQuickFix(IDocument document, ISonarLintIssuable issuable, List<MarkerQuickFix> qfs, QuickFixDto rpcQuickFix) {
+    var qf = new MarkerQuickFix(rpcQuickFix.message());
+    for (var edits : rpcQuickFix.fileEdits()) {
+      var fileWithEdit = SonarLintUtils.findFileFromUri(edits.target());
+      if (!issuable.equals(fileWithEdit)) {
+        SonarLintLogger.get().debug("Quick fix on multiple files is not supported yet: " + rpcQuickFix.message());
         return;
       }
+      // TODO
```

Comment:
Get rid of this, please 😄

Reply:
I removed the dead code, I just left a question instead